cudamapper/src/main.cu

```diff
@@ -129,65 +129,74 @@ int main(int argc, char* argv[])
         std::cerr << "NOTE - Since query and target files are same, activating all_to_all mode. Query index size used for both files." << std::endl;
     }
 
-    std::unique_ptr<claragenomics::io::FastaParser> query_parser = claragenomics::io::create_fasta_parser(query_filepath);
+    std::shared_ptr<claragenomics::io::FastaParser> query_parser = claragenomics::io::create_kseq_fasta_parser(query_filepath);
     int32_t queries                                              = query_parser->get_num_seqences();
 
-    std::unique_ptr<claragenomics::io::FastaParser> target_parser = claragenomics::io::create_fasta_parser(target_filepath);
-    int32_t targets                                               = target_parser->get_num_seqences();
+    std::shared_ptr<claragenomics::io::FastaParser> target_parser;
+    if (all_to_all)
+    {
+        target_parser = query_parser;
+    }
+    else
+    {
+        target_parser = claragenomics::io::create_kseq_fasta_parser(target_filepath);
+    }
+
+    int32_t targets = target_parser->get_num_seqences();
 
     std::cerr << "Query " << query_filepath << " index " << queries << std::endl;
     std::cerr << "Target " << target_filepath << " index " << targets << std::endl;
 
     // Data structure for holding overlaps to be written out
     std::mutex overlaps_writer_mtx;
 
-    struct query_target_range
+    struct QueryTargetRange
```

Comment:
Would `QueryTargetsRange` be a better name?

Reply:
yes. good point. fixed